pkg/client/gen.go

```diff
@@ -43,18 +46,8 @@ type templateValues struct {
 
 // GenerateManifest fills in a template with a Sonobuoy config
 func (c *SonobuoyClient) GenerateManifest(cfg *GenConfig) ([]byte, error) {
-	if cfg.Image != "" {
-		cfg.Config.WorkerImage = cfg.Image
-	}
-
-	if cfg.ImagePullPolicy != "" {
-		cfg.Config.ImagePullPolicy = cfg.ImagePullPolicy
-	}
-
-	if cfg.Namespace != "" {
-		cfg.Config.Namespace = cfg.Namespace
-	}
-
+	// Provide defaults but don't overwrite any customized configuration.
+	mergo.Merge(cfg.Config, config.New())
```

Comment:
we actually wanted Merge, not MergeWithOverrides. The reason for this is that a partially configured `cfg.Config` will create zero values for everything that isn't provided. Merge will then pull in defaults for all of those zero values based on the non zero values provided by `config.New()`